src/future.cc

```diff
@@ -44,10 +53,25 @@ scmd::query_result scmd::future::get_result() {
 
 scmd::prepared_query scmd::future::get_prepared() {
     const CassPrepared *result = cass_future_get_prepared(this->_future);
-    if(result == nullptr) {
-        //TODO: Maybe throw error result and/or error message?
+    if (result == nullptr) {
+        // TODO: Maybe throw error result and/or error message?
         scmd_internal::throw_on_cass_error(cass_future_error_code(this->_future));
     }
     return scmd::prepared_query(result);
 }
 
+void scmd::future::set_callback_fast(scmd::future::callback_type_fast f) {
+    this->cb->set_callback_fast(f);
+    scmd_internal::throw_on_cass_error(cass_future_set_callback(this->_future, scmd_internal::future_callback::callback_fn_fast, static_cast<void*>(this->cb)));
+}
+
+void scmd::future::set_callback_fast(scmd::future::callback_type_fast_bound f, void *arg) {
+    this->cb->set_callback_fast(f, arg);
+    scmd_internal::throw_on_cass_error(cass_future_set_callback(this->_future, scmd_internal::future_callback::callback_fn_fast_bound, static_cast<void*>(this->cb)));
+}
+
+void scmd::future::set_callback(const callback_type_universal &f) {
+    this->cb->set_callback(f);
+    scmd_internal::throw_on_cass_error(cass_future_set_callback(this->_future, scmd_internal::future_callback::callback_fn, static_cast<void*>(this->cb)));
+}
```

Comment:
Here the order should be fixed, too, similarly to the modification suggested above.